db-engine/src/main/java/com/splicemachine/db/impl/sql/GenericStatement.java

```diff
@@ -687,8 +685,14 @@ else if (newMergeJoinString.equals("forced"))
                 preparedStmt.compilingStatement=false;
                 preparedStmt.notifyAll();
             }
-            DataDictionaryCache.fromTableTriggerDescriptor.remove();
-            DataDictionaryCache.fromTableTriggerSPSDescriptor.remove();
+            TriggerReferencingStruct.fromTableTriggerDescriptor.remove();
+            TriggerReferencingStruct.fromTableTriggerSPSDescriptor.remove();
+            // Communicate to the immediate parent statement if its child
+            // contains a FROM TABLE clause.
+            if (BoundAndOptimizedStatement != null)
```

Comment:
Can you rename `BoundAndOptimizedStatement` in all places to firstLowerCase?

Reply:
Done.